src/Spout/Writer/ODS/Manager/WorksheetManager.php

```diff
@@ -155,14 +155,32 @@ public function addRow(Worksheet $worksheet, Row $row)
      * @throws InvalidArgumentException If a cell value's type is not supported
      * @return string
      */
-    private function applyStyleAndGetCellXML(Cell $cell, Style $rowStyle, $currentCellIndex, $nextCellIndex)
+    private function applyStyleAndGetCellXML(Cell $cell, Style &$rowStyle, $currentCellIndex, $nextCellIndex)
```

Comment:
I'm not a fan of using reference passing. I see this as a side effet that can easily be missed or go out of control.
This also is a good sign that this method does too many things and it's a great time to refactor it. Maybe compute the rowStyle first, then the cell style, then return the XML?

Reply:
Indeed I can avoid reference, if we register rowStyle first in `addNonEmptyRow`, before calling `applyStyleAndGetCellXML`.
But register rowStyle even if is not used is not a bad thing ?

Reply:
Well, registering a style increases the memory consumption. If the styles being registered as always or almost always the same, then the impact on the memory is limited so that could be a valid option.

Another option is to split what's done here in 2:
1. Compute and register the cell/row style
2. Get the XML for the registered style

Reply:
I created `RegisteredStyle` like `ManagedStyle` in order to avoid reference.  I also split `applyStyleAndGetCellXML` to `applyStyleAndRegister` and `getCellXMLWithStyle`